src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Herbivore && obj.hidden === false) {
+      obj.health -= 50;
+    }
+
+    if (!obj.health) {
+      const indexOfAnimal = Animal.alive.indexOf(obj);
+
+      if (indexOfAnimal > -1) {
+        Animal.alive.splice(indexOfAnimal, 1);
+      }
+    }
```

Comment:
[Checklist](https://github.com/mate-academy/js_herbivores_and_carnivores/blob/master/checklist.md) 2nd item, use filter instead to improve code performance